eth/handler.go

```diff
@@ -444,11 +444,9 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 			// Irrelevant of the fork checks, send the header to the fetcher just in case
 			headers = pm.fetcher.FilterHeaders(headers, time.Now())
 		}
-		if len(headers) > 0 || !filter {
```

Comment:
This is explicitly how the fetcher works, it keeps anything sent to it, and forwards the rest to the downloader. If the fetcher kept the header delivered, and forwards an empty slide to the downloader, the downloader will abort the sync thinking the peer doesn't have any more headers available.